### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Mzuha**

### CODE REVIEW

File: src/main/java/mate/web/filter/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if (driverId == null && allowedUrls.contains(req.getServletPath())) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        if (driverId == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
+    }
```
Comment (Tamara-Paskalova) [inline src/main/java/mate/web/filter/AuthenticationFilter.java:47]:
```suggestion
        if (driverId == null && !allowedUrls.contains(req.getServletPath())) {
            resp.sendRedirect("/login");
            return;
        }

        filterChain.doFilter(req, resp);
    }
```

File: src/main/java/mate/controller/car/GetMyCurrentCarsController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@WebServlet(urlPatterns = "/cars/my")
+public class GetMyCurrentCarsController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private final CarService carService = (CarService) injector.getInstance(CarService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        HttpSession session = req.getSession();
+        Long id = (Long) session.getAttribute("driverId");
```
Comment (Tamara-Paskalova) [inline src/main/java/mate/controller/car/GetMyCurrentCarsController.java:24]: you should make "driver_id" a constant field here and everywhere you use it

**Review by Tamara-Paskalova — CHANGES_REQUESTED** [review]

**Review by Tamara-Paskalova — APPROVED** [review]

File: src/main/java/mate/dao/impl/DriverDaoImpl.java [hunk 3]
```diff
@@ -101,11 +104,30 @@ public boolean delete(Long id) {
         }
     }
 
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT id, login, password, name,"
+                + " license_number FROM drivers WHERE login = ?";
```
Comment (Sofasmile) [inline src/main/java/mate/dao/impl/DriverDaoImpl.java:109]: and is_deleted = false?

File: src/main/java/mate/service/impl/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Invalid login or password");
+        }
+        if (driver.get().getPassword().equals(password)) {
```
Comment (Sofasmile) [inline src/main/java/mate/service/impl/AuthenticationServiceImpl.java:22]: can we combine these two cheks into one if block?

File: src/main/resources/init_db.sql [hunk 5]
```diff
@@ -4,6 +4,10 @@ SET FOREIGN_KEY_CHECKS = 0;
 -- ----------------------------
 -- Table structure for drivers
 -- ----------------------------
+DROP SCHEMA `taxi`;
+CREATE SCHEMA `taxi` DEFAULT CHARACTER SET utf8;
+USE `taxi`;
+
```
Comment (Sofasmile) [inline src/main/resources/init_db.sql:10]: let's add login and password to init_db file?

**Review by Sofasmile — CHANGES_REQUESTED** [review]
great!

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**